.github/workflows/deploy-preview.yml

```
@@ -89,6 +101,105 @@ jobs:
+      - name: Configure AWS credentials for URL pre-calculation
+        if: steps.decision.outputs.should-deploy == 'true'
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_ARN }}
+          aws-region: ${{ env.AWS_REGION }}
+          role-session-name: GitHubActions-PreviewDeploy-URLPrecalc-${{ github.run_id }}
+
+      - name: Pre-calculate deployment URLs
+        id: url-precalc
+        if: steps.decision.outputs.should-deploy == 'true'
+        run: |
```

actionlint (1.7.7)

"github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions for more details

(expression)